Style without layout of todo list

JungDaewook opened the pull request:
[구현 사항]
- 레이아웃을 제외한 스타일(색상, 글자 크기) 구현.

[중점 리뷰 사항]
- css 코드에서 중복된 코드가 많은데, 상위 태그에서 설정해줌으로써 코드 중복을 피할 수 있는 방법이 있는지
- 본인은 최대한 상위 태그에 설정하려고 노력했으나, 혹시 더 좋은 방법이 있는지 궁금

Commit:
코드 가독성을 위해 레이아웃과 스타일 부분의 코드 분리

Commit:
최상단 제목의 스타일 구현.

Commit:
todoListContentHeader 디자인 구현 완료
1. js 파일에서 추가버튼과 전체삭제 버튼에 id 부여
2. todoItem 입력창과 추가, 전체삭제 버튼 디자인 구현.

Commit:
#todoList 안의 요소들 디자인 구현 완료.
1. js의 innerHTML 코드에서 삭제 버튼의 id 부여
2. todoList 안의 요소들 디자인 구현.

JungDaewook commented on functionsOfTodoList.js:
```diff
@@ -14,7 +14,7 @@ function addTodoItem()  {
   todoItem.setAttribute("id", todoItemId);
   todoItem.appendChild(textNode);
-  todoItem.innerHTML += "<div class = 'strikeAndRemoveButton'><input type = 'button' id = 'strikeButton' onclick = 'strikeTodoItem(\""+todoItemId+"\")' value = '완료'/><input type = 'button' onclick = 'removeTodoItem(\""+todoItemId+"\")' value = '삭제'/></div>"; 
+  todoItem.innerHTML += "<div class = 'strikeAndRemoveButton'><input type = 'button' id = 'strikeButton' onclick = 'strikeTodoItem(\""+todoItemId+"\")' value = '완료'/><input type = 'button' id = 'removeButton' onclick = 'removeTodoItem(\""+todoItemId+"\")' value = '삭제'/></div>"; 
```
[설명]
삭제 버튼의 디자인을 구현하기 위해서 'removeButton' 이라는 id 부여

JungDaewook replied:
[설명]
삭제 버튼의 디자인을 구현하기 위해서 'removeButton' 이라는 id 부여

JungDaewook commented on Todo_List.html:
```diff
@@ -12,9 +12,9 @@ <h1>Todo-List</h1>
-                        <input type = 'button' value = '전체 삭제' onclick = 'removeAllTodoItem()'/>
+                    <div class = "todoListContentHeaderButtons">
+                        <input type = 'button' id = 'addButton' value = '추가' onclick = 'handleAddButton()' />
+                        <input type = 'button' id = 'removeAllButton' value = '전체 삭제' onclick = 'removeAllTodoItem()'/>
```
[설명]
디자인 구현을 위해 추가 버튼과 전체삭제 버튼에 id 부여

JungDaewook commented:
[질문]
커밋의 단위를 큼직큼직함 컨테이너들로 나눴는데 더 쪼개는 것이 좋을까요?

Commit:
bodyContainer의 설정 일부를 body로 이동
1. bodyContainer와 body 사이에 하얀색으로 margin이 잡히는 문제 해결
2. todoItem을 많이 추가했을 때 아래 부분에 bodyContainer의 설정이 해결되지 않는 문제 해결

JungDaewook reviewed (commented):
1. bodyContainer와 body 사이에 하얀색으로 margin이 잡히는 문제 해결
2. todoItem을 많이 추가했을 때 아래 부분에 bodyContainer의 설정이 해결되지 않는 문제 해결
<img width="1728" alt="스크린샷 2022-07-16 오전 12 23 39" src="https://user-images.githubusercontent.com/40848888/179257391-cc73026e-0580-474b-b352-0532d3a091d2.png">
<img width="847" alt="스크린샷 2022-07-16 오전 12 25 32" src="https://user-images.githubusercontent.com/40848888/179257418-682f45a2-c19c-41ed-bd6a-11a7933ef8dd.png">

Commit:
추가, 완료 버튼에도 삭제, 전체삭제 버튼과 같이 hover 추가

Commit:
불필요한 코드 제거 및 css 파일 줄 나눔

unpo88 commented on styleOfTodoList.css:
```diff
@@ -1,38 +1,126 @@
+
+.todoListContentHeaderButtons input {
+    flex : auto;
+
```
@JungDaewook 

이 부분은 개행이 들어간 이유가 있을까요?

JungDaewook replied:
[답변 / 질문]
가독성을 높이려는 목적으로 전체적인 레이아웃을 담당하는 코드와 세세한 스타일을 담당하는 코드를 분리하려고 개행을 넣었어요. 아래의 개행들도 같은 이유입니다. 혹시 잘못되거나 잘 사용하지 않는 방식일까요? 만약 잘못된 방식이라면 수정하도록 하겠습니다.

unpo88 replied:
@JungDaewook 

[질문]

잘못되거나 잘 사용하지 않는 방식인지 혹은 잘 사용하는 방식인지를 찾아보려면 어떻게 찾아보면 좋을까요?

JungDaewook replied:
[https://velog.io/@rlatmdgns94/guide](url)
[답변]
위의 링크에서 css 작성 가이드라인 자료를 찾아본 후 사용하지 않는 방식이라는 것을 알게 돼서 불필요한 개행을 아래 커밋에서 제거했어요.
https://github.com/JungDaewook/Todo-JavaScript/pull/16/commits/eb247e60264b9790f2d26eb00c96a02dc2f0ace1

unpo88 commented on styleOfTodoList.css:
```diff
@@ -1,38 +1,126 @@
+
+#todoInputText {
+    flex : auto;
+
```
@JungDaewook 

이 부분은 개행이 들어간 이유가 있을까요?

JungDaewook replied:
[답변]
가독성을 높이려는 목적으로 전체적인 레이아웃을 담당하는 코드와 세세한 스타일을 담당하는 코드를 분리하려고 개행을 넣었어요. 혹시 잘못되거나 잘 사용하지 않는 방식일까요?

unpo88 commented on styleOfTodoList.css:
```diff
@@ -1,38 +1,126 @@
 .todoListContentContainer {
-    width : 600px;
     flex-shrink : 1;
+
```
@JungDaewook 

이 부분도 개행이 들어간 이유가 있을까요?

JungDaewook replied:
[답변]
가독성을 높이려는 목적으로 전체적인 레이아웃을 담당하는 코드와 세세한 스타일을 담당하는 코드를 분리하려고 개행을 넣었어요. 혹시 잘못되거나 잘 사용하지 않는 방식일까요?

unpo88 commented on styleOfTodoList.css:
```diff
@@ -1,38 +1,126 @@
+
 .bodyContainer h1 {
     text-align : center;
+
```
@JungDaewook 

이 부분은 개행이 들어간 이유가 있을까요?

unpo88 commented on styleOfTodoList.css:
```diff
@@ -1,38 +1,126 @@
 #todoList li {
     display : flex;
     justify-content : space-between;
+
```
@JungDaewook 

여기도 개행이 들어간 이유가 있을까요?

unpo88 reviewed (changes requested):
@JungDaewook 

> css 코드에서 중복된 코드가 많은데, 상위 태그에서 설정해줌으로써 코드 중복을 피할 수 있는 방법이 있는지
> 본인은 최대한 상위 태그에 설정하려고 노력했으나, 혹시 더 좋은 방법이 있는지 궁금

[질문]
1. 현재 작성된 코드를 기준으로 중복된 코드는 어떠한 부분을 이야기하는 것일까요?
2. CSS 중복 제거를 위해 찾아본 자료가 있을까요? 있다면 어떤 자료를 살펴보았을까요?

Commit:
불필요한 컨테이너 제거 및 css 코드 수정

JungDaewook commented on Todo_List.html:
```diff
@@ -5,22 +5,17 @@
     <title>Todo-List</title>
 </head>
 <body>
-    <div class = "bodyContainer">
```
[코드 수정]
bodyContainer 제거

unpo88 replied:
@JungDaewook 

이렇게 모든 작업마다 코멘트를 달아놓은 이유가 있을까요?
리뷰어가 이 코멘트를 통해 어떤 정보를 얻기를 기대하는 것일까요?

`bodyContainer 제거`라는 정보는 코드에 나와있는 것을 그대로 이야기한 것 같은데
코멘트는 코드에 드러나지 않거나 특별한 부분에 대해서 리뷰를 받고 싶은 부분이 있거나
리뷰어가 이해하기 어려운 부분에 대해서 코멘트가 달려야할 것 같은데

이 경우에는 위에 제가 나열한 경우에 해당되는 것 같지 않아서요 🤔 
오히려 리뷰어에게 불필요한 정보처럼 느껴집니다.

unpo88 replied:
아래 코멘트가 달린것들을 보니 모두 동일하네요.

우선 Resolve 처리하겠습니다. 왜 그렇게 작성했는지, 어떠한 것을 기대했는지 설명 부탁드립니다 🙏

JungDaewook replied:
[답변]
커밋을 살펴보니 코드가 통째로 삭제됐다가 추가된 것처럼 나타나 있어서 리뷰어가 바뀐 것을 파악하기 어려울 수도 있겠다는 판단 하에 세세하게 코멘트를 남겼어요.

지금 보니 저도 리뷰어가 충분히 코드를 보고 이해할 수 있을 것 같다는 생각이 들어요. 다음부터는 주의하도록 할게요!

JungDaewook commented on Todo_List.html:
```diff
@@ -5,22 +5,17 @@
-    <div class = "bodyContainer">
-        <h1>Todo-List</h1>
-        <div class = "todoListMainContainer">
-            <div class = "todoListNavContainer"></div>
```
[코드 수정]
todoListNavContainer 제거

unpo88 replied:
@JungDaewook 

위 코멘트와 동일합니다.

JungDaewook commented on Todo_List.html:
```diff
@@ -5,22 +5,17 @@
                 </div>
-                <ul id = 'todoList'></ul>
             </div>
-            <div class = "todoListAdContainer"></div>
```
[코드 수정]
todoListAdContainer 제거

JungDaewook commented on styleOfTodoList.css:
```diff
@@ -16,11 +13,7 @@ body {
-
-.todoListNavContainer {
-    flex-grow : 1;
+    justify-content : center;
```
[코드 추가]
nav / ad 컨테이너가 없어서 가운데 정렬을 해주기 위한 코드 추가

Commit:
css 코드 띄어쓰기 통일

JungDaewook commented on styleOfTodoList.css:
```diff
@@ -1,119 +1,115 @@
+    background-color: rgb(0, 81, 255);
+    display: flex;
+    flex-direction: column;
 }
```
[코드 수정 / 질문]
- 코드 수정
css 코드에서 많이 사용하는 방식으로 띄어쓰기를 수정했어요. 이유는 구글링을 해 봤는데 잘 안 나왔어요. 

- 질문
변수 같은 것을 할당 할때는 왼쪽의 변수와 오른쪽의 상수를 구분하기 위해 `int a = 3;`과 같이 등호의 왼쪽과 오른쪽에 띄어쓰기를 추가하는데, css의 경우에는 할당을 하는 것이 아닌 왼쪽의 속성 값(display)이 flex라는 것만 알면 되기 때문에 띄어쓰기를 : 의 오른쪽에만 추가하는 것이 맞나요?

unpo88 reviewed (changes requested):
@JungDaewook 

상단에 다양한 코멘트를 남겨놓았어요.

JungDaewook commented:
> @JungDaewook
> 
> > css 코드에서 중복된 코드가 많은데, 상위 태그에서 설정해줌으로써 코드 중복을 피할 수 있는 방법이 있는지
> > 본인은 최대한 상위 태그에 설정하려고 노력했으나, 혹시 더 좋은 방법이 있는지 궁금
> 
> [질문]
> 
> 1. 현재 작성된 코드를 기준으로 중복된 코드는 어떠한 부분을 이야기하는 것일까요?
> 2. CSS 중복 제거를 위해 찾아본 자료가 있을까요? 있다면 어떤 자료를 살펴보았을까요?

[답변]
1. border-radius 과 같은 속성들은 같은 값을 가지는 요소들이 많아서 각 요소들의 스타일에 모두 들어가는 중복이 발생했어요.
2. CSS 중복 뿐만이 아니라 CSS 코드 작성 가이드라인에 대해 찾아보았어요. 
https://velog.io/@rlatmdgns94/guide

Commit:
불필요한 개행 제거

Commit:
li 요소에 marker를 추가하기 위한 css 코드 수정.

Commit:
marker 위치 조정을 위한 코드 삭제 및 추가
1. list-style: none; 코드로 marker 삭제
2. ::before 를 이용해 위치를 조정할 수 있는 marker(속성은 before인) 생성.

JungDaewook commented:
[코드 수정]
https://github.com/JungDaewook/Todo-JavaScript/pull/16/commits/225e834134d3b89917b40289c0f3ab29dc629496
위 커밋으로 marker 위치 조정에 성공했어요.

JungDaewook commented:
[질문 / merge 여부]
이제 모든 코멘트에 대한 답변과 수정이 완료된 것 같은데 merge해도 될까요?

Commit:
innerHTML 코드 가독성 문제 해결
1. 완료, 취소, 삭제 버튼의 innerHTML 내용을 return하는 함수 생성

JungDaewook commented:
[리팩토링]
https://github.com/JungDaewook/Todo-JavaScript/pull/16/commits/524934111e528f6d0a4a1acb2e403b969678cc5a

[코드 수정]
- 위 커밋에서 innerHTML 코드의 가독성을 위해 코드를 추가하고 수정했어요.

[세부 사항]
- 완료, 취소, 삭제 버튼의 innerHTML 내용을 반환하는 함수를 만들어 가로로 길었던 코드를 축소하고 중복되는 코드들을 함수로 대체했어요.

JungDaewook commented on styleOfTodoList.css:
```diff
@@ -69,15 +69,31 @@ body h1 {
     font-size: 20px;
     border-radius: 20px;
-    list-style-position: inside;
+    position: relative;
```
[질문]
position에 대한 충분한 조사가 있었나요?

JungDaewook replied:
[답변]
https://github.com/JungDaewook/Todo-JavaScript/pull/16/commits/389566187fd05814b4121ea3fe4a8eb0c0d7c52a
불필요한 코드라고 생각되어 위 커밋에서 삭제했어요.

JungDaewook commented on styleOfTodoList.css:
```diff
@@ -69,15 +69,31 @@ body h1 {
+    border-radius: 100%;
+    background-color: black;
+    margin: 8px;
+    margin-left: 10px;
```
[코멘트]
margin 과 margin-left 는 한번에 쓸 수 있지 않을까요?

JungDaewook replied:
[수정 완료]
https://github.com/JungDaewook/Todo-JavaScript/pull/16/commits/4059c4f0cf224d32626bc94217cdf03af8bdbfac
위 커밋에서 축약했어요.

JungDaewook commented on styleOfTodoList.css:
```diff
@@ -69,15 +69,31 @@ body h1 {
     min-height: 300px;
     border-radius: 15px;
     padding-left: 0;
+    list-style-position: inside;
```
[질문]
이 코드는 꼭 필요한 코드일까요?

Commit:
코멘트에 따라 불필요한 코드 수정 및 축약
1. 불필요한 코드 제거
2. 축약 가능한 코드 축약
3. li 안의 요소들 수직 방향으로 가운데 정렬

Commit:
불필요한 코드 제거

JungDaewook commented on styleOfTodoList.css:
```diff
@@ -80,20 +80,18 @@ body h1 {
 
 #todoList li::before {
     content: "";
     position: absolute;
```
[문제 발생 / 해결]

- [문제 상황]
`position: absolute`라고 설정하지 않으면 밑 사진과 같이 before 태그가 flex의 레이아웃에 참여하는 현상이 있었어요.
<img width="599" alt="스크린샷 2022-08-02 오후 5 27 20" src="https://user-images.githubusercontent.com/40848888/182329723-1063120f-eebf-4925-9ae7-27611eac7059.png">

- [해결]
`position: absolute`로 설정해주니 before 태그가 flex의 레이아웃에 참여하지 않아 밑 사진과 같이 의도한대로 구현할 수 있었어요.
<img width="581" alt="스크린샷 2022-08-02 오후 5 33 50" src="https://user-images.githubusercontent.com/40848888/182330377-98c59e41-0127-45d5-938c-62d16658248b.png">

- [참고 자료]
밑의 링크를 통해 position을 absolute로 설정해주면 해당 설정이 적용된 태그가 flex의 레이아웃에 참여하지 않는다는 사실을 알게 됐어요.
https://chenhuijing.com/blog/flexbox-and-absolute-positioning/#%F0%9F%A7%97%E2%80%8D%E2%99%80%EF%B8%8F

Commit:
fonctionsOfTodoList.js 파일에서 버튼 관련 코드 중복 문제 해결
1. 버튼을 만들기 위한 innerHtml text를 div 태그로 감싸는 함수 구현
2. 함수 내에서 버튼을 만들기 위한 innerHtml text를 변수로 설정.

Commit:
wrapTextForInnerHtmlWithDivTag함수의 함수명을 함수 역할을 더 명확하게 드러내는 함수명으로 변경

Commit:
구현과정에서 실수로 남아있던 코드 제거

JungDaewook commented on functionsOfTodoList.js:
```diff
@@ -2,6 +2,7 @@ const MAX_TODO_ITEM_COUNT = 20;
 let todoItemCount = 0;
 let todoItemIdIndex = 0;
 let isTodoListEmpty = true;
+let todoItemId = returningTodoItemId();
```
[불필요한 코드 제거]
https://github.com/JungDaewook/Todo-JavaScript/pull/16/commits/cb96fc33936590f3807a4a84ddc38fdebf0010bd
구현과정에서 나온 불필요한 코드를 실수로 남겨놓아서 위 커밋에서 제거했어요. 커밋을 남길 때 변경사항들을 다시 한번 확인하는 습관을 길러야겠다고 생각했어요.

JungDaewook commented:
[코드 수정]
https://github.com/JungDaewook/Todo-JavaScript/pull/16/commits/a86f387868c60f19fb399e747af37519f329bf7f

- [수정 사항]
버튼 관련 코드가 중복되는 문제를 위 커밋으로 해결했어요.
1. 버튼을 만들기 위한 innerHtml text를 div 태그로 감싸는 함수 구현
2. 함수 내에서 버튼을 만들기 위한 innerHtml text를 변수로 설정.
